src/Arcus.Observability.Telemetry.Serilog.Enrichers/CorrelationInfoEnricher.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using Arcus.Observability.Correlation;
+using Arcus.Observability.Telemetry.Core;
+using GuardNet;
+using Serilog.Core;
+using Serilog.Events;
+
+namespace Arcus.Observability.Telemetry.Serilog.Enrichers
+{
+    /// <summary>
+    /// Enriches the log events with the correlation information.
+    /// </summary>
+    public class CorrelationInfoEnricher : ILogEventEnricher
+    {
+        private readonly ICorrelationInfoAccessor _correlationInfoAccessor;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CorrelationInfoEnricher"/> class.
+        /// </summary>
+        /// <param name="correlationInfoAccessor">The accessor implementation for the <see cref="CorrelationInfo"/> model.</param>
+        public CorrelationInfoEnricher(ICorrelationInfoAccessor correlationInfoAccessor)
+        {
+            Guard.NotNull(correlationInfoAccessor, nameof(correlationInfoAccessor));
+
+            _correlationInfoAccessor = correlationInfoAccessor;
+        }
+
+        /// <summary>Enrich the log event.</summary>
+        /// <param name="logEvent">The log event to enrich.</param>
+        /// <param name="propertyFactory">Factory for creating new properties to add to the event.</param>
+        public void Enrich(LogEvent logEvent, ILogEventPropertyFactory propertyFactory)
+        {
+            Guard.NotNull(logEvent, nameof(logEvent));
+            Guard.NotNull(propertyFactory, nameof(propertyFactory));
+
+            CorrelationInfo correlationInfo = _correlationInfoAccessor.CorrelationInfo;
+
+            if (correlationInfo is null)
+            {
+                return;
+            }
+
+            if (!String.IsNullOrEmpty(correlationInfo.OperationId))
+            {
+                LogEventProperty property = propertyFactory.CreateProperty(ContextProperties.Correlation.OperationId, correlationInfo.OperationId);
+                logEvent.AddPropertyIfAbsent(property);
+            }
+
+            if (!String.IsNullOrEmpty(correlationInfo.TransactionId))
+            {
+                LogEventProperty property = propertyFactory.CreateProperty(ContextProperties.Correlation.TransactionId, correlationInfo.TransactionId);
+                logEvent.AddPropertyIfAbsent(property);
+            }
```

Comment:
This was what I was wondering actually - How would you approach it to get messaging correlation info then? Wouldn't it make more sense to provide an enricher per app type then?

Reply:
For the rest, 👌

Reply:
Hmm, maybe we can indeed make some inject point in the enricher where the user can send for a custom correlation accessor a custom function. 🤔
I'll come up with something.

Reply:
Updated with function now... is it something you could agree upon?

Reply:
Instead of a function I would make it a method which can be overriden; this is more the approach used in .NET

Reply:
But fixes the issue indeed, thanks!

Reply:
😆 I guess it's the functional programmer in me to imm think of functions.